daemon/src/main/java/bisq/daemon/grpc/GrpcDisputeAgentsService.java

```diff
@@ -36,4 +48,22 @@ public void registerDisputeAgent(RegisterDisputeAgentRequest req,
             exceptionHandler.handleException(cause, responseObserver);
         }
     }
+
+    final ServerInterceptor[] interceptors() {
+        Optional<ServerInterceptor> rateMeteringInterceptor = rateMeteringInterceptor();
+        return rateMeteringInterceptor.map(serverInterceptor ->
+                new ServerInterceptor[]{serverInterceptor}).orElseGet(() -> new ServerInterceptor[0]);
+    }
+
+    final Optional<ServerInterceptor> rateMeteringInterceptor() {
+        CallRateMeteringInterceptor defaultCallRateMeteringInterceptor =
+                new CallRateMeteringInterceptor(new HashMap<>() {{
+                    // You can only register mainnet dispute agents in the UI.
+                    // Do not limit devs' ability to register test agents.
+                    put("registerDisputeAgent", new GrpcCallRateMeter(1, SECONDS));
+                }});
+
+        return getCustomRateMeteringInterceptor(coreApi.getConfig().appDataDir, this.getClass())
+                .or(() -> Optional.of(defaultCallRateMeteringInterceptor));
+    }
```

Comment:
These `rateMeteringInterceptor` methods seem repetitive. Are they degradeable to constants? The unique items seem to be the arguments to `put()` and `this`; might be possible to factor out the rest of the logic into a common method. I'm not really opposed to leaving this unchanged, just something to consider.

Reply:
Not sure how to fully answer this, but we cannot create a super class for these Grpc*Service classes, if that's what you are suggesting.  A Grpc service class must extend a class generated from a proto def.

Something else you might not realize... but ought to know:  a unique interceptor is bound to each Grpc*Service, and each interceptor has it's own unique set up rate meters.

Reply:
But we could create an interceptor factory class to factor out some of the repetition.

Reply:
Commit c996240 only factors out repeated calls to 'new CallRateMeteringInterceptor'.

I want to leave the `rateMeteringInterceptor` methods there because I expect there will be a need for other types of interceptors in the future, and I want it to be clear that the calling `final ServerInterceptor[] interceptors()` method should get them from a `<type>Interceptor` method.